SpatialGDK/Source/SpatialGDK/Public/Utils/Interest/ResultTypes.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/SpatialClassInfoManager.h"
+#include "Schema/Interest.h"
+
+/**
+ * This class contains static functionality to build the required result types for query based interest.
+ *
+ * If the required result type will contain generated components, then there is a dependency on the ClassInfoManager to retrieve those.
+ * Otherwise the result types are built using the predefined component sets in SpatialConstants.
+ */
+
+namespace SpatialGDK
+{
+
+class SPATIALGDK_API ResultTypes
```

Comment:
Is it worth renaming it to something like `InterestQueryResultTypes`?

Reply:
Changed to `ResultTypeCreators` based on the conversation in the other thread.

Reply:
I like the idea, that the naming explains what the role is. However, it doesn't seem obvious to me, what the Result Type is itself. I could accept it though, since this functionality is internal to the `SpatialClassInfoManager`.